bufferevent_sock.c

```diff
@@ -498,6 +498,41 @@ bufferevent_connect_getaddrinfo_cb(int result, struct evutil_addrinfo *ai,
 }
 
 int
+bufferevent_socket_set_src_address(struct bufferevent *bev,
+    const struct sockaddr *addr, int socklen)
+{
+	struct bufferevent_private *bufev_p =
+	    EVUTIL_UPCAST(bev, struct bufferevent_private, bev);
+
+	evutil_socket_t fd;
+	int r = 0;
+	int result = -1;
```

Comment:
That's the way the code already was (in `bufferevent_socket_connect()` where I was borrowing from) and I was trying to be "low touch".  I could change the code but it would end up adding a few lines...  Since I would have something like:

```
    result = evutil_socket_bind(fd, addr, socklen);
    if (result < 0) {
        result = -1;
        goto done;
    }

    result = 0;
```

instead.  As I see it, `r` is the return value from the system call.  `result` is the return value of this function itself.  We could get rid of `r` but then `result` would be overloaded.